Sources/ComposableArchitecture/Documentation.docc/Articles/MigratingTo1.6.md

```diff
@@ -0,0 +1,638 @@
+}
+```
+
+Then you would have used ``SwitchStore`` and ``CaseLet`` in the view like this:
+
+```swift
+SwitchStore(store) {
+  switch $0 {
+  case .activity:
+    CaseLet(/Feature.State.activity, action: Feature.Action.activity) { store in
+      ActivityView(store: store)
+    }
+  case .settings:
+    CaseLet(/Feature.State.settings, action: Feature.Action.settings) { store in
+      SettingsView(store: store)
+    }
+  }
+}
+```
+
+This can now be updated to use a vanilla `switch` and `case` in the view:
+
+```swift
+switch store.state {
+case .activity:
+  if let store = store.scope(state: \.activity, action: \.activity) {
+    ActivityView(store: store)
+  }
+case .settings:
+  if let store = store.scope(state: \.settings, action: \.settings) {
+    SettingsView(store: store)
+  }
+}
+```
+
+## Replacing navigation view modifiers with SwiftUI modifiers
+
+The library has shipped many navigation view modifiers that mimic what SwiftUI provides, but are
+tuned specifically for driving navigation from a ``Store``. All of these view modifiers can be
+updated to instead use the vanilla SwiftUI version of the view modifier, and so the modifier that
+ship with this library are now soft-deprecated.
+
+For example, if your feature's reducer looks roughly like this:
+
+```swift
+@Reducer
+struct Feature {
+  @ObservableState
+  struct State {
+    @PresentationState var child: Child.State?
+  }
+  enum Action {
+    case child(PresentationAction<Child.State>
+  }
+  var body: some ReducerOf<Self> { /* ... */ }
+}
+```
+
+Then previously you would drive a sheet presentation from this feature like so:
+
+```swift
+.sheet(store: store.scope(state: \.$child, action: \.child)) { store in
+  ChildView(store: store)
+}
+```
+
+You can now replace `sheet(store:)` with the vanilla SwiftUI modifier, `sheet(item:)`. First you
+must hold onto the store in your view in a bindable manner, either using `@State`:
+
+```swift
+@State var store: StoreOf<Feature>
+```
+
+…or using the `@Bindable` property wrapper:
+
+```swift
+@Bindable var store: StoreOf<Feature>
+```
+
+Then you can use `sheet(item:)` like so:
+
+```swift
+.sheet(item: $store.scope(state: \.child, action: \.child)) { store in
+  ChildView(store: store)
+}
+```
+
+Note that the state key path is simply `state: \.child`, and not `state: \.$child`. The projected
+value of the presentation state is no longer needed.
+
+This also applies to popovers, full screen covers, and navigation destinations.
+
+Also, if you are driving navigation from an enum of destinations, then currently your code may
+look something like this:
+
+```swift
+.sheet(
+  store: store.scope(
+    state: \.$destination.editForm,
+    action: \.destination.editForm
+  )
+) { store in
+  ChildView(store: store)
+}
+```
+
+This can now be changed to this:
+
+```swift
+.sheet(
+  item: $store.scope(
+    state: \.destination?.editForm,
+    action: \.destination.editForm
+  )
+) { store in
+  ChildView(store: store)
+}
+```
```

Comment:
The above currently doesn't seem to work with the `@Bindable` property wrapper as its type will be `Bindable<Store<State, Action>>` and that type doesn't have a `scope` method defined that would return a `Binding`. 

Copying the implementation of the corresponding method from the `Binding` extension sort of does the job, but there's no `transaction` property defined on `Bindable` and I am not familiar enough with this topic to suggest a proper solution.

```
@available(iOS 17.0, *)
extension Bindable {
  public func scope<State: ObservableState, Action, ChildState, ChildAction>(
    state: KeyPath<State, ChildState?>,
    action: CaseKeyPath<Action, PresentationAction<ChildAction>>
  ) -> Binding<Store<ChildState, ChildAction>?>
  where Value == Store<State, Action> {
    let isInViewBody = PerceptionLocals.isInPerceptionTracking
    return Binding<Store<ChildState, ChildAction>?>(
      get: {
        // TODO: Is this right? Should we just be more forgiving in bindings?
        PerceptionLocals.$isInPerceptionTracking.withValue(isInViewBody) {
          self.wrappedValue.scope(
            state: state,
            action: action.appending(path: \.presented)
          )
        }
      },
      set: {
        if $0 == nil, self.wrappedValue.stateSubject.value[keyPath: state] != nil {
//          Bindable doesn't have a transaction property --> self.transaction($1).wrappedValue.send(action(.dismiss))
        }
      }
    )
  }
}
```

Reply:
Hey @0xcefaedfe, thanks for bringing this up. It is true that `@Bindable` doesn't work quite as seamlessly as we hoped (or as the docs suggest). For now you can just use `@State` and we will look into `@Bindable` in more depth soon, or perhaps introduce a different tool.

Reply:
@0xcefaedfe We've pushed helpers for `@Bindable` to this branch now. Thanks for letting us know!